lib/api/networkset.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"github.com/projectcalico/libcalico-go/lib/api/unversioned"
+	"github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// NetworkSet contains a set of arbitrary IP sub-networks/CIDRs that share labels to
+// allow rules to refer to them via selectors.  The labels of NetworkSets are in the same
+// "namespace" as those for endpoints.  NetworkSets are useful to refer to groups
+// of "external" non-Calico workloads/hosts.
+type NetworkSet struct {
```

Comment:
Just to close the loop, I think we agreed over slack that the current name was OK given that there's only one label namespace so, if we add new types of "target" later, they'll just union in with what we've got.